transformer/transformer.go

```diff
@@ -0,0 +1,289 @@
+package transformer
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const startHighlightTag string = "<strong>"
+const endHighlightTag string = "</strong>"
+
+// Transformer represents an instance of the ResposeTransformer interface
+type Transformer struct {
+}
+
+type searchResponse struct {
+	Count        int           `json:"count"`
+	Took         int           `json:"took"`
+	Items        []contentItem `json:"items"`
+	ContentTypes []contentType `json:"content_types"`
+}
+type contentItem struct {
+	Description description `json:"description"`
+	Type        string      `json:"type"`
+	URI         string      `json:"uri"`
+	Matches     matches     `json:"matches,omitempty"`
+}
+type description struct {
```

Comment:
I would generally put fields in alphabetical order, so it is easier to locate a field in a long list (including swagger spec); unless there is another reason to the logical order


fyi - this order will be preserved in the json response

Reply:
Code updated to make (most) fields alphabetically ordered.  I have left some unordered in cases where it seems better. Eg. summary counts etc.